### PR TITLE
truncate points_[found,acknowledged]_24h to 24 hours at each signage point

### PR DESCRIPTION
**Pull request by altendky**
While working on https://github.com/Chia-Network/chia-blockchain/pull/7905 I noticed that `points_found_24h` contained a lot more than 24 hours worth of point data.  This was already noted at https://github.com/Chia-Network/chia-blockchain/pull/7611#pullrequestreview-710163459.  This change truncates the list at each signage point to provide reasonable resolution of the list.

Draft for:
- [x] Check if Mariano had anything particular in mind for fixing this.
- [x] Wait 24 hours to see if it is really working...
- [x] Check into removing whatever the GUI does to address this at least for the points since they seem correct.
  - It looks like [`removeOldPoints()`](accumulator) is the compensation in the GUI and it is applied to both [`points_acknowledged_24h` and `points_found_24h`](https://github.com/Chia-Network/chia-blockchain-gui/blob/3bbef783557fb1aab267f341cfe8f7b86b509c5e/src/util/normalizePoolState.ts#L7-L8)
  - Even though I might be able to hack it out, I am not a ~~JS~~ typescript dev so if that should be removed it would probably be good to call in the reinforcements.
- [x] Move the code to a function and do this for points_acknowledged_24h as well.
- [x] Find a better place than when a proof is found.  If it takes awhile between proofs being found then the data can grow well past 24 hours.

### CODE REVIEW

**Comment by altendky** [comment]
@mariano54, over at https://github.com/Chia-Network/chia-blockchain/pull/7611#pullrequestreview-710163459 you commented about fixing this yourself.  Did you have a particular approach in mind or should I continue with this?  I am certainly fine dropping this if you wanted to handle it yourself.  I just figured I'd put something concrete here to look at in case it is any good.

**Comment by mariano54** [comment]
This is fine, I haven't had time to work on it. Thank you for the PR!

The only thing I would say, is that we should not do it right before processing signage point, as that will slow down farming.
We should put it when the pool itself returns success for a partial

**Comment by altendky** [comment]
The move makes sense in terms of timing, thanks for the suggestion.  Do you specifically want to only update in case of the pool accepting the points?  It seems like it should happen regardless of success or any sort of failure.  I believe you are suggesting the area referenced below.  My first thought is to put it in a `finally:`.

https://github.com/Chia-Network/chia-blockchain/blob/d380957460d2a8f15436119192e62b7e7c25001b/chia/farmer/farmer_api.py#L220-L254

**Comment by altendky** [comment]
Looks like macOS CI is broken.  Kick it.  If it fails again I'll give it some more time.

**Comment by hoffmang9** [comment]
If you cherry pick deac3ad3cc788f89d089b7d48cca8fe6c866e471 or merge main back in - you'll get the MacOS runner fixes. GH made changes to the runner last night...

**Comment by hoffmang9** [comment]
Ah. @cmmarslender forgot to make that conditional for PRs like yours.

**Comment by cmmarslender** [comment]
> Ah. @cmmarslender forgot to make that conditional for PRs like yours.

Sorry about that! I have a PR open to fix this issue

**Comment by altendky** [comment]
Well, now that I have it 24 hours to make sure it worked I realized that only doing this when a proof has been found means that data older than 24 hours won't get trimmed.  For small farms this could mean hours of extra data being included.  I'll try to find another spot that also stays out of the path from an incoming signage point to the harvester and back to the pool.

**Comment by altendky** [comment]
While waiting to see what this looks like tomorrow night after 24 hours...

It seems like tying this to signage points at all isn't "correct".  Let's say your system isn't doing well to the point that signage points aren't being handled, you would think that your 24hr points would indicate some issue.  But, perhaps this is enough of an improvement to be ok.  I don't know what sort of opinions you all have on "better" vs. "correct" on non-critical paths like this.

I could imagine it being handled at the point of handling the API request even.  Or on it's own timer.  Some hybrid with a timer to kick in if the signage point doesn't trigger it first.  Etc, etc.

In terms of performance concerns, I think it is ok to do this after sending messages to the harvesters?  The harvesters of course still have to respond and we may have to pass a proof back to the network, but it seems like we ought to have a little time in the middle there, especially if the harvester(s) have had to lookup proofs.   If more efficiency is desired then perhaps it would be good to switch from lists to deques to get O(1) removals.  Or, perhaps I am still being too cavalier with putting code _anywhere_ at all between receiving a signage point and reporting a proof/partial.  Then again, it's async.  Other stuff can get mixed in anyways even between each harvester getting notified.  This at least controls where in the process it will be done.

**Comment by altendky** [comment]
I'm not used to the big-long-sequence tests that I found for the farmer API so I just tossed a small test on the end for now.  I'll try to read back through the whole test to get a better feel for how we handle this area.

File: chia/farmer/farmer_api.py [hunk 1]
```diff
@@ -27,6 +27,16 @@
 from chia.util.ints import uint32, uint64
 
 
+def strip_old_entries(pairs: Sequence[Tuple[float, Any]], before: float):
```
Comment (mariano54) [inline chia/farmer/farmer_api.py:30]: I would add return type annotation for clarity

**Review by mariano54 — DISMISSED** [review]

**Comment by mariano54** [comment]
Performance is not a big concern, since the harvester already has the SP and is working on disk, and the dictionaries will not be very large.

File: chia/farmer/farmer_api.py (hunk 1 above)
Comment (altendky) [reply chia/farmer/farmer_api.py]: Quite so, thanks for pointing it out.  Fixed.

**Review by mariano54 — APPROVED** [review]

**Pull request merged**